pkg/controller/jobs/trainjob/trainjob_controller.go

```diff
@@ -274,6 +274,18 @@ func (t *TrainJob) RunWithPodSetsInfo(ctx context.Context, podSetsInfo []podset.
 	if t.Annotations == nil {
 		t.Annotations = map[string]string{}
 	}
+	// Filter out the existing overrides that were added by Kueue
+	// (identified by the presence of the PodSetLabel).
+	// This makes the function idempotent, preventing duplicate overrides
+	// if the update operation is retried.
+	var userOverrides []kftrainer.PodTemplateOverride
```

Comment:
Same here, this is making production code changes. We need to undestand the impact well and have dedicated PR.

Reply:
There's no need to do this since this function is called at start/resume job time, and the Kueue injected overrides are be removed by the `RemovePodSetsInfo` function at job stop time. If this update fails, the original `TrainJob` won't be modified and the reconciler will reuse it when retrying.

Reply:
sgtm, I'm wondering then how this was triggered on our e2e tests as reported by @j-skiba here: https://github.com/kubernetes-sigs/kueue/pull/8269#discussion_r2623428350

The code look harmless so I'm ok to include it as a safeguard. Still, I'm also unclear what is the problematic scenario.